crates/stackable-versioned-macros/src/gen/venum/variant.rs

```diff
@@ -0,0 +1,273 @@
+use std::{collections::BTreeMap, ops::Deref};
+
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+use syn::{Ident, Variant};
+
+use crate::{
+    attrs::variant::VariantAttributes,
+    gen::{
+        chain::{BTreeMapExt, Neighbors},
+        common::{remove_deprecated_variant_prefix, ContainerVersion, ItemStatus, VersionChain},
+    },
+};
+
+#[derive(Debug)]
+pub(crate) struct VersionedVariant {
+    chain: Option<VersionChain>,
+    inner: Variant,
+}
+
+// TODO (@Techassi): Figure out a way to be able to only write the following code
+// once for both a versioned field and variant, because the are practically
+// identical.
+
+impl VersionedVariant {
+    pub(crate) fn new(variant: Variant, variant_attrs: VariantAttributes) -> Self {
+        // NOTE (@Techassi): This is straight up copied from the VersionedField
+        // impl. As mentioned above, unify this.
+        if let Some(deprecated) = variant_attrs.common.deprecated {
```

Comment:
As mentioned in another comment, I think we can get attributes from `variant`.

```suggestion
    pub(crate) fn new(variant: Variant) -> Self {
        let variant_attrs = VariantAttributes::from_variant(&variant)?;
        // NOTE (@Techassi): This is straight up copied from the VersionedField
        // impl. As mentioned above, unify this.
        if let Some(deprecated) = variant_attrs.common.deprecated {
```

Reply:
Changed in de1cde3